tests/examples/CMakeLists.txt

```diff
@@ -24,15 +24,13 @@ execute_process(COMMAND lua -e "p = package.path:gsub(';', '\\\\;'); io.stdout:w
 if(NOT SOURCE_DIR AND ${CMAKE_CURRENT_SOURCE_DIR} MATCHES "/tests/examples")
     get_filename_component(TOP_SOURCE_DIR
                            "${CMAKE_CURRENT_SOURCE_DIR}/../.." ABSOLUTE)
-    # Used by .luacov.
-    set(ENV{SOURCE_DIRECTORY} ${TOP_SOURCE_DIR})
 else()
     set(TOP_SOURCE_DIR ${CMAKE_SOURCE_DIR})
 endif()
 
 if (DO_COVERAGE)
   execute_process(
-    COMMAND lua -e "require('luacov.runner')('${TOP_SOURCE_DIR}/.luacov')"
+    COMMAND env "SOURCE_DIRECTORY=${TOP_SOURCE_DIRECTORY}" lua -e "require('luacov.runner')('${TOP_SOURCE_DIR}/.luacov')"
```

Comment:
Currently it fails to generate coverage:

> build/luacov.stats.out does not exist.

Likely related to this change I would guess (also above maybe).

Reply:
My guess is that the missing coverage happens because `make` does not run the example tests. The later run that is now there thanks to @actionless has a different build dir, so will place the `luacov.stats.out` file somewhere completely different...

Reply:
so, should SOURCE_DIRECTORY here be changed for that or should example build be changed to put output to a different location?

Reply:
This PR shouldn't change anything. Things should still work as before.

Which means that the example tests should be run during `make` (with which I mean our wrapper-Makefile that runs `cmake` and `make` in a build dir). For some reason this does not happen on Travis, but it does happen on the two computers that I tested it with. And debugging things on Travis is annoying. :-/